docs/examples/requests.md

```diff
@@ -0,0 +1,31 @@
+!!! warning "🚧 Work in Progress"
+    This page is a work in progress. More examples will be added soon.
+
+## `httpx` requests
+
+[`httpx`](https://www.python-httpx.org/) is a HTTP client for Python 3 with synchronous and asynchronous APIs.
+In the below example, we query the [JSONPlaceholder API](https://jsonplaceholder.typicode.com/) to get a user's data and validate it with a Pydantic model.
+
+```python test="skip"
+import httpx
+
+from pydantic import BaseModel, EmailStr
+
+
+class User(BaseModel):
+    id: int
+    name: str
+    email: EmailStr
+
+
+url = 'https://jsonplaceholder.typicode.com/users/1'
+
+response = httpx.get(url)
+response.raise_for_status()
+
+user = User.model_validate(response.json())
```

Comment:
It may be worth calling out here the use of a `TypeAdapter` for cases where the response isn't a single model (e.g., if the response is an array). I know you had an example of that somewhere earlier in this PR but my experience is that you usually want to use a TypeAdapter with web clients just so that it can be super consistent across all types of data, and the nice thing is that type adapter works even if you use a non-pydantic struct to store the data (e.g., a dataclass)

Reply:
Good call. I've added an additional example that queries the `/users/` endpoint to get a list of users with a `TypeAdapter`.